fiftyone/utils/activitynet.py

```diff
@@ -0,0 +1,1175 @@
+                random.shuffle(ids)
+
+            if requested_num is not None:
+                requested_sample_ids = ids[:requested_num]
+            else:
+                requested_sample_ids = ids
+
+            return requested_sample_ids
+
+        return []
+
+    def _download_requested_samples(
+        self, ids, class_samples, shuffle, requested_num, num_workers, split
+    ):
+        if (requested_num is None or requested_num) and ids:
+            ids = sorted(ids)
+            if shuffle:
+                random.shuffle(ids)
+
+            downloaded_ids = self._separate_versions_and_attempt_to_download(
+                ids, class_samples, requested_num, num_workers, split,
+            )
+
+            return downloaded_ids
+
+        return []
+
+    def _separate_versions_and_attempt_to_download(
+        self, ids, samples_info, num_samples, num_workers, split,
+    ):
+        all_a100_ids = self.a100_info.all_sample_ids
+        a100_ids = [i for i in ids if i in all_a100_ids]
+        a200_ids = list(set(ids) - set(a100_ids))
+
+        downloaded_ids = []
+        num_a100_ids = len(a100_ids)
+        remaining_samples = num_samples
+
+        if num_a100_ids:
+            logger.info("Downloading videos...")
+            if num_samples is None:
+                num_to_download = num_a100_ids
+            else:
+                num_to_download = min(num_a100_ids, num_samples)
+
+            downloaded_ids, a100_errors = self._attempt_to_download(
+                self.a100_info.data_dir(split),
+                a100_ids,
+                samples_info,
+                num_to_download,
+                num_workers,
+            )
+            if remaining_samples is not None:
+                remaining_samples -= len(downloaded_ids)
+
+            self._merge_and_write_errors(
+                a100_errors, self.a100_info.error_path(split)
+            )
+
+        if remaining_samples:
+            logger.info("Downloading ActivityNet-200-specific videos...")
+            a200_downloaded_ids, a200_errors = self._attempt_to_download(
+                self.a200_info.data_dir(split),
+                a200_ids,
+                samples_info,
+                remaining_samples,
+                num_workers,
+            )
+            downloaded_ids.extend(a200_downloaded_ids)
+            self._merge_and_write_errors(
+                a200_errors, self.a200_info.error_path(split)
+            )
+
+        return downloaded_ids
+
+    def _attempt_to_download(
+        self, videos_dir, ids, samples_info, num_samples, num_workers
+    ):
+        download_urls = []
+        download_paths = []
+        for sample_id in ids:
+            sample_info = samples_info[sample_id]
+            download_path = os.path.join(videos_dir, "v_%s.mp4" % sample_id)
+            download_urls.append(sample_info["url"])
+            download_paths.append(download_path)
+
+        downloaded, errors = fouy.download_youtube_videos(
```

Comment:
I didn't investigate the code in detail, but in your implementation, do you rely on the downloaded video exactly matching `download_path` above? Because it might not; eg it could be a different format if no MP4 streams are available.

Reply:
Thanks for pointing that out. No, it doesn't exactly need to match. MP4 streams are just preferred.